src/faking.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Utils for generating fake data
+ *
+ * Not very complete. For now I'm injecting rooms into a live client like this:
+ * mxMatrixClientPeg.get()._syncApi._processSyncResponse({}, {
+ *   rooms: { join: Faking.randomly(Faking.Rooms(100)) }
+ * })
+ */
+
+/*
+ * Irritatingly, Javascript does not provide a seedable RNG.  A seedable RNG is very
+ * helpful when you want reproduceable results.
+ */
+interface Xorshift128pState {
+    a: bigint,
+    b: bigint,
+}
+const MAX = (BigInt(1) << BigInt(64)) - BigInt(1);
+export function xorshift128p(s: Xorshift128pState): { state: Xorshift128pState, value: number } {
+    let t: bigint = s.a;
+    t ^= (t << BigInt(23)) & MAX;
+    t ^= (t >> BigInt(17));
+    t ^= s.b ^ (s.b >> BigInt(26));
+    const state = { a: s.b, b: t };
+    const value = Number((t + s.a) & MAX) / Number(MAX);
+    return { state, value };
+}
```

Comment:
This needs comments, and probably documenting as a seedable RNG instead of some complicated math function.